src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt

```diff
@@ -423,6 +426,40 @@ sealed class ImportInfo {
 
 data class ImportCandidate(val importItem: ImportItem, val info: ImportInfo)
 
+private val RsPath.namespaceFilter: (RsQualifiedNamedElement) -> Boolean get() = when (context) {
+    is RsTypeElement -> { e ->
+        when (e) {
+            is RsEnumItem,
+            is RsStructItem,
+            is RsTraitItem,
+            is RsTypeAlias -> true
+            else -> false
+        }
+    }
+    is RsPathExpr -> { e ->
+        when (e) {
+            is RsConstant,
+            is RsFunction -> true
+            is RsFieldsOwner -> e.blockFields == null
```

Comment:
There is a case when we typing a struct literal and want to import struct before typing `{`, i.e.
```rust
let a = Foo/*caret*/
```
This `Foo` will be parsed as `RsPathExpr`, but we may want to import a struct.
```rust
let a = Foo { ... };
```
In short, I think this `e.blockFields == null` is not necessary

Reply:
> In short, I think this e.blockFields == null is not necessary

I think we should understand such things to more precise suggestions in quick fix but your case with typing is reasonable so let's drop check of fields type for now

> P.S. + RsTypeAlias
Do you mean to add `RsTypeAlias` as yet another element in `when`? But I can't come up with case when you can use type alias in path expr. Maybe you mean such cases for `RsStructLiteral` branch?

Reply:
> RsTypeAlias

I meant the same case with incomplete struct literal, where we refer not a `RsFieldsOwner` directly, but a `RsTypeAlias`

Reply:
Oh, okay:)
I've fixed it too